lib/AssemblyUtil/TypeToFasta.py

```diff
@@ -0,0 +1,80 @@
+
+
+import os
+
+from installed_clients.DataFileUtilClient import DataFileUtil
+from installed_clients.MetagenomeUtilsClient import MetagenomeUtils
+from AssemblyUtil.AssemblyToFasta import AssemblyToFasta
+
+from shutil import copyfile
+
+class TypeToFasta:
+
+    def __init__(self, callback_url, scratch, ws_url):
+        self.ws_url = ws_url
+        self.callback_url = callback_url
+        self.scratch = scratch
+        self.dfu = DataFileUtil(callback_url)
+        self.mgu = MetagenomeUtils(callback_url)
+
+    def type_to_fasta(self, ctx, ref_lst):
```

Comment:
I think to take type checking into account I would have to create a util function just for that... Because I can get the object type version, sure that's easy but then once I have that how to compare to the latest type version for that object? Is the version I have out of date? Is the version I have new?

If I just put all the KBase object type versions into a dictionary and then check type versions against that it would work.... and just having that function around for object version checking might be useful.

Reply:
and since different genome versions would have different routes to a fasta, it would be useful to just do a if statement based on version. But in order to do that I would need a way to check what version I have relative to the old and new KBase object versions that exist.

Reply:
You can def do that, there are workspace functions to get the list of versions for a type. But really, I'm not sure that's necessary - what you'd do, *if* this seems like the right thing to do, is look at the version and then run specific code based on that version. If you don't have code for that version, you can either throw and error or just try stuff until an approach works, which can be dangerous depending on the version changes.

The other approach is the just try stuff approach (e.g. try key A, if nothing there, then try Key B to get the ref, etc), which also can be dangerous depending on the version changes.

What you decide to do depends on how confident you are that the part of the object you need is not going to change in future versions.

Reply:
fixed this with:
self.ws.get_objects2({'objects': \
                            [{"ref": genome_upa, 'included' : ['/assembly_ref/','/contigset_ref/']}]}) \
                            ['data'][0]['data']
and an if statement